clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/EcsTargetGroupCacheClient.java

```diff
@@ -0,0 +1,134 @@
+    EcsTargetGroup ecsTargetGroup = objectMapper.convertValue(targetGroupAttributes, EcsTargetGroup.class);
+    return ecsTargetGroup;
+  }
+
+  private List<EcsTargetGroup> convertToTargetGroup(Collection<Map<String, Object>> targetGroupAttributes) {
+    List<EcsTargetGroup> ecsTargetGroups = new ArrayList<>();
+
+    for (Map<String, Object> attributes : targetGroupAttributes) {
+      ecsTargetGroups.add(convertToTargetGroup(attributes));
+    }
+
+    return ecsTargetGroups;
+  }
+
+
+  private Set<Map<String, Object>> fetchLoadBalancerAttributes(Collection<String> targetGroupKeys) {
+    Set<CacheData> targetGroups = fetchTargetGroups(targetGroupKeys);
+
+    Set<CacheData> targetGroupsWithAtLeastOneLoadBalancer = targetGroups
+      .stream()
+      .filter(targetGroup -> targetGroup.getRelationships().get("loadBalancers") != null
+        && targetGroup.getRelationships().get("loadBalancers").size() > 0)
+      .collect(Collectors.toSet());
+
+    Set<Map<String, Object>> targetGroupAttributes = targetGroupsWithAtLeastOneLoadBalancer.stream()
+      .map(lb -> lb.getAttributes())
+      .collect(Collectors.toSet());
+
+    return targetGroupAttributes;
+  }
+
+
+  private Set<Map<String, Object>> retrieveTargetGroups(Set<String> targetGroupsAssociatedWithLoadBalancers) {
+    Collection<CacheData> targetGroupCache = cacheView.getAll(TARGET_GROUPS.getNs(), targetGroupsAssociatedWithLoadBalancers);
+    Set<Map<String, Object>> targetGroupAttributes = targetGroupCache
+      .stream()
+      .map(lb -> lb.getAttributes())
+      .collect(Collectors.toSet());
+
+    return targetGroupAttributes;
+  }
+
+
+  private Set<String> inferAssociatedTargetGroups(Set<CacheData> loadBalancers) {
+    Set<String> targetGroupsAssociatedWithLoadBalancers = new HashSet<>();
+
+    for (CacheData loadBalancer : loadBalancers) {
+      Collection<String> relatedTargetGroups = loadBalancer.getRelationships().get("targetGroups");
+      if (relatedTargetGroups != null && relatedTargetGroups.size() > 0) {
+        for (String targetGroupArn : relatedTargetGroups) {
+          targetGroupsAssociatedWithLoadBalancers.add(targetGroupArn);
+        }
+      }
+    }
+    return targetGroupsAssociatedWithLoadBalancers;
+  }
+
+  private Set<CacheData> fetchTargetGroups(Collection<String> targetGroupKeys) {
+    return cacheView
+      .getAll(TARGET_GROUPS.getNs(), targetGroupKeys, RelationshipCacheFilter.include(LOAD_BALANCERS.getNs()))
+      .stream()
+      .collect(Collectors.toSet());
+  }
+
+
+  private String extractAccount(EcsTargetGroup ecsTargetGroup) {
+    return null; // Implement this based on ecsTargetGroup.getId(); -->  example value: aws:targetGroups:continuous-delivery:us-west-2:clouddriver-dev:vpc-bb8f37dd
+  }
+
+  private String extractRegion(EcsTargetGroup ecsTargetGroup) {
+    return null; // Implement this based on ecsTargetGroup.getId(); -->  example value: aws:targetGroups:continuous-delivery:us-west-2:clouddriver-dev:vpc-bb8f37dd
+  }
```

Comment:
Do these two methods need to be implemented before the class will work?

Reply:
Nope - removed.